xbmc/cores/VideoPlayer/Interface/Addon/InputStreamConstants.h

```diff
@@ -8,6 +8,35 @@
 
 #pragma once
 
-#define STREAM_PROPERTY_INPUTSTREAMCLASS  "inputstreamclass" /*!< @brief the name of the inputstream add-on that should be used by Kodi to play the stream denoted by STREAM_PROPERTY_STREAMURL. Leave blank to use Kodi's built-in playing capabilities or to allow ffmpeg to handle directly set to STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG. */
-#define STREAM_PROPERTY_ISREALTIMESTREAM "isrealtimestream" /*!< @brief "true" to denote that the stream that should be played is a realtime stream. Any other value indicates that this is not a realtime stream.*/
-#define STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG  "inputstream.ffmpeg" /*!< @brief special value for STREAM_PROPERTY_INPUTSTREAMCLASS to use ffmpeg to directly play a stream URL. */
+/*!
+ * @brief the name of the inputstream add-on that should be used by Kodi to
+ * play the stream denoted by STREAM_PROPERTY_STREAMURL. Leave blank to use
+ * Kodi's built-in playing capabilities or to allow ffmpeg to handle directly
+ * set to STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG.
+ */
+#define STREAM_PROPERTY_INPUTSTREAMCLASS "inputstreamclass"
+
+/*!
+ * @brief Property to define instance identification string.
+ *
+ * This then becomes given to addon during his creation.
+ * Used on inputstream addon instance to identify about something special,
+ * e.g. inputstream instance inside PVR and want to separated between
+ * channel or recording.
+ *
+ * To be used together with @ref STREAM_PROPERTY_INPUTSTREAMCLASS
+ */
+#define STREAM_PROPERTY_INPUTSTREAMCLASS_IDENT "inputstreamclass.ident"
```

Comment:
I would use "ID" instead of "IDENT" and ".id" instead of ".ident". I've never seen that we use ident for shortening identity, but we use id a lot.

Reply:
Becomes renamed to
```cpp
#define STREAM_PROPERTY_INPUTSTREAM_DATA_ID "inputstream.dataid"
```